_pytest/main.py

```diff
@@ -210,7 +210,9 @@ def __getitem__(self, key):
             return self.parent.keywords[key]
 
     def __setitem__(self, key, value):
-        self._markers[key] = value
+        from _pytest.mark import MarkDecorator
+        if not isinstance(self._markers.get(key), MarkDecorator):
```

Comment:
I'm not particularly comfortable with the fix either, so I tried to check for the duplicate key higher in the stack. I looked at `PyCollector._genfunctions`, for example, where I thought that the `keywords` argument passed to `Function` could be empty when `callspec.params['name'] == callspec.params['arg']` but that seemed even more convoluted. Any suggestion would be most welcome.